src/factorFinder/FactorComputationCoordinatorImpl.java

```diff
@@ -0,0 +1,66 @@
+package com.example.factorfinder;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FactorComputationCoordinatorImpl
+    implements FactorComputationCordinator {
+  private final FactorComputeEngine
+      computeEngine; // Reference to the compute engine
+  private final FactorDataStore dataStore; // Reference to the data store
+
+  // Constructor to initialize the compute engine and data store
+  public FactorComputationCoordinatorImpl(
+      FactorComputeEngine computeEngine, FactorDataStore dataStore) {
+    this.computeEngine = computeEngine;
+    this.dataStore = dataStore;
+  }
+
+  @Override
+  public FactorComputeResult compute(FactorComputeRequest request) {
+    // Correlates to step 2a. Here we receive requests from the user to start the computation
+    InputConfig inputConfig =
+        request.getInputConfig(); 
+
```

Comment:
This is still using the interface instead of the implementation of the InputConfig (FileInputConFig), I'm fixing it right now  in eclipse to see if anything will be messed up